pkg/tracing/jaeger/config_yaml.go

```diff
@@ -4,165 +4,140 @@
-	}
-
-	if cfg.SamplerManagerHostPort != "" {
-		sc.SamplingServerURL = cfg.SamplerManagerHostPort
-	}
-
-	if cfg.SamplerMaxOperations != 0 {
-		sc.MaxOperations = cfg.SamplerMaxOperations
+// getAgentEndpointOptions returns Jaeger options populated with agent related options.
+func getAgentEndpointOptions(config Config) []otel_jaeger.AgentEndpointOption {
+	var endpointOptions []otel_jaeger.AgentEndpointOption
+	endpointOptions = append(endpointOptions, otel_jaeger.WithAgentHost(config.AgentHost))
+	endpointOptions = append(endpointOptions, otel_jaeger.WithAgentPort(strconv.Itoa(config.AgentPort)))
+
+	// This option was part of the Jaeger config was JAEGER_REPORTER_ATTEMPT_RECONNECTING_DISABLED.
+	if config.ReporterDisableAttemptReconnecting {
+		endpointOptions = append(endpointOptions, otel_jaeger.WithDisableAttemptReconnecting())
+		if config.ReporterAttemptReconnectInterval != 0 {
+			endpointOptions = append(endpointOptions, otel_jaeger.WithAttemptReconnectingInterval(config.ReporterAttemptReconnectInterval))
+		}
 	}
 
-	if cfg.SamplerRefreshInterval != 0 {
-		sc.SamplingRefreshInterval = cfg.SamplerRefreshInterval
+	if config.ReporterLogSpans {
+		var logger *log.Logger
+		endpointOptions = append(endpointOptions, otel_jaeger.WithLogger(logger))
 	}
 
-	return sc
+	return endpointOptions
 }
 
-// reporterConfigFromConfig creates a new ReporterConfig based on the YAML Config.
-func reporterConfigFromConfig(cfg Config) (*config.ReporterConfig, error) {
-	rc := &config.ReporterConfig{}
-
-	if cfg.ReporterMaxQueueSize != 0 {
-		rc.QueueSize = cfg.ReporterMaxQueueSize
-	}
-
-	if cfg.ReporterFlushInterval != 0 {
-		rc.BufferFlushInterval = cfg.ReporterFlushInterval
-	}
-
-	if cfg.ReporterLogSpans {
-		rc.LogSpans = cfg.ReporterLogSpans
+// getSamplingFraction returns the sampling fraction based on the sampler type.
+// Ref: https://www.jaegertracing.io/docs/1.35/sampling/#client-sampling-configuration
+func getSamplingFraction(samplerType string, samplingFactor float64) float64 {
+	if samplerType == "const" {
+		if samplingFactor > 1 {
+			return 1.0
+		} else if samplingFactor < 0 {
+			return 0.0
+		}
+		return math.Round(samplingFactor) // Returns either 0 or 1 for values [0,1].
+	} else if samplerType == "probabilistic" {
+		return samplingFactor
+	} else if samplerType == "ratelimiting" {
+		return math.Round(samplingFactor) // Needs to be an integer.
 	}
+	return samplingFactor
+}
 
-	if cfg.Endpoint != "" {
-		u, err := url.ParseRequestURI(cfg.Endpoint)
-		if err != nil {
-			return nil, errors.Wrapf(err, "cannot parse endpoint=%s", cfg.Endpoint)
-		}
-		rc.CollectorEndpoint = u.String()
-		user := cfg.User
-		pswd := cfg.Password
-		if user != "" && pswd == "" || user == "" && pswd != "" {
-			return nil, errors.Errorf("you must set %s and %s parameters together", cfg.User, cfg.Password)
+func getSampler(samplerType string, samplingFraction float64, parentConfig ParentBasedSamplerConfig) tracesdk.Sampler {
```

Comment:
@matej-g I have added the considerations for sampler type here.
I'm not sure how to add rate limiting samplers yet.
PTAL and let me know if this is okay, thanks!